src/main/java/org/hydev/mcpm/client/commands/entries/InfoController.java

```diff
@@ -0,0 +1,101 @@
+package org.hydev.mcpm.client.commands.entries;
+
+import com.google.common.collect.Streams;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.text.WordUtils;
+import org.hydev.mcpm.client.database.PluginTracker;
+import org.hydev.mcpm.utils.ColorLogger;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Controller for the info use case
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-22
+ */
+public record InfoController(PluginTracker tracker)
+{
+    private static final int LEN_LEFT = 12;
+    private static final int LEN_RIGHT = 60;
+    private static final String KV_PRE = "&7> &f";
+    private static final String KV_SEP = " : &6";
+    private static final int LEN_INDENT = LEN_LEFT + ColorLogger.lengthNoColor(KV_PRE + KV_SEP);
+
+    /**
+     * Format a kv pair if value isn't null, return empty string if value is null
+     *
+     * @param key Key
+     * @param rawValue Value
+     * @return Formatted pair
+     */
+    private static String formatPair(String key, @Nullable Object rawValue)
+    {
+        if (rawValue == null) return "";
+
+        // If a map is passed in, only show its keys
+        if (rawValue instanceof Map<?,?> v)
```

Comment:
I still think it's fairly strange that you're not using overloading for this but sure... 🤔